docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,295 @@
+include::./attributes.adoc[]
+= {project-name} - Asynchronous messages between beans
+
+{project-name} allows different beans to interact using asynchronous messages, enforcing loose-coupling.
+The messages are sent to _virtual addresses_.
+It offers 3 types of delivery mechanism:
+
+- point-to-point - send the message, one consumer receives it. If several consumers listen to the address, a round robin is applied;
+- publish/subscribe - publish a message, all the consumers listening to the address are receiving the message;
+- request/reply - send the message and expect a response. The receiver can respond to the message in an asynchronous-fashion
+
+All these delivery mechanism are non-blocking, and are providing one of the fundamental brick to build reactive applications.
+
+NOTE: The asynchronous message passing feature allows replying to messages which is not supported by Reactive Messaging.
+However, it is limited to single-event behavior (no stream) and to local messages.
+
+== Installing
+
+This mechanism uses the Vert.x EventBus, so you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vertx"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Consuming events
+
+To consume events, use the `io.quarkus.vertx.ConsumeEvent` annotation:
+
+[source, java]
+----
+package org.acme.vertx;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class GreetingService {
+
+    @ConsumeEvent                           // <1>
+    public String consume(String name) {    // <2>
+        return name.toUpperCase();
+    }
+}
+----
+<1> If not set, the address is the fully qualified name of the bean, for instance, in this snippet it's `org.acme.vertx.GreetingService`.
+<2> The method parameter is the message body. If the method returns _something_ it's the message response.
+
+// TODO Revisit this once the blocking PR is integrated - https://github.com/quarkusio/quarkus/pull/1646
```

Comment:
FYI https://github.com/quarkusio/quarkus/pull/1646 was already merged.

Reply:
Yes, but we want to publish this guide before the actual release of Quarkus. Quarkus release may be delayed due to a few regressions.

Reply:
Np, I'm just saying it's there ;-).